### ISSUE TITLE
Ignore variants in glossary checks

### ISSUE THREAD

**Issue by nijel** [WeblateOrg/weblate#11883]
### Discussed in https://github.com/WeblateOrg/weblate/discussions/11726

<div type='discussions-op-text'>

<sup>Originally posted by **Akemi-Madoka** May 28, 2024</sup>
How should Variants in the Glossary work with check_glossary in Translation checks?
Example:
"タバナ" => "Tabana"
"タバナ・オミノ" => "Tabana・omino"
"タバナさま" => "Tabana-Sama"
"タババちゃん" => "Tababa-Chan"
"魔王の刃" => "Demon blade"
These four Japanese terms refer to the same person in different contexts, and I think they should fit using the Variants feature.
However, when we created "タバナ" along with the other four Variants in the Glossary, all entries containing any one of the Variants failed to pass check_glossary, no matter which Variant's translation it was translated to.
In summary, a translation of one Variant needs to satisfy all the translations of its Variants to pass check_glossary, which is very confusing for us.
Example: "ちなみにワタシは タバナ。　タバナ・オミノ。"
In this case, the corresponding translation needs to include "Tabana・omino," "Tabana-Sama," "Tababa-Chan," and "Demon blade" to pass check_glossary.</div>

**Comment by nijel** [comment]
The variants should be only displayed in the sidebar and not influence checks or machine translation (if the source string does not match).
